userspace/libsinsp/sinsp.cpp

```diff
@@ -1020,6 +1022,26 @@ void sinsp::restart_capture_at_filepos(uint64_t filepos)
 	}
 }
 
+uint32_t sinsp::get_max_n_proc_loopkups()
+{
+	uint32_t ret = m_max_n_proc_lookups;
+
+	static const uint32_t startup = sinsp_utils::get_steady_clock_ts();
+
+	if(m_check_bump_max_n_proc_lookups)
+	{
+		auto now = sinsp_utils::get_steady_clock_ts();
+		m_check_bump_max_n_proc_lookups = ((now - startup) < BUMP_MAX_N_PROC_LOOKUPS_DURATION_IN_SEC);
+		if(m_check_bump_max_n_proc_lookups)
+		{
+			ret = (STARTUP_MAX_N_PROC_LOOKUPS*(BUMP_MAX_N_PROC_LOOKUPS_DURATION_IN_SEC - (now - startup)) + m_max_n_proc_lookups)/BUMP_MAX_N_PROC_LOOKUPS_DURATION_IN_SEC;
```

Comment:
This calculation is hard to read, maybe split it into several parts, e.g.

```
// note this is signed
int64_t remaining_lookup_bump_time = BUMP_MAX_N_PROC_LOOKUPS_DURATION_IN_SEC - (now - startup);
if(remaining_lookup_bump_time > 0)
{
    // the linear drop is more visible now IMHO
    // also, your code includes m_max_n_proc_lookups in the linear equation instead of adding it on top, is that intentional?
    uint32_t boost = STARTUP_MAX_N_PROC_LOOKUPS * remaining_lookup_bump_time / BUMP_MAX_N_PROC_LOOKUPS_DURATION_IN_SEC;
    return m_max_n_proc_lookups + boost;
}
return m_max_n_proc_lookups;
```

On an unrelated note, early returns instead of nested `if`s?

Reply:
The way you wrote the method has a drawback. You need to calculate each time the variable now even if *m_check_bump_max_n_proc_lookups* has been already set to false. I have cascading `if` precisely to avoid this.
Moreover, I belong to the party "Early return is evil". Worst than pineapple on pizza.
Anyway I'll try to make the formula clearer splitting boost and asymptotic value as you suggested